test-ts/proxy-deploy.test.ts

```diff
@@ -1,80 +1,124 @@
 import { expect } from "chai";
 import hre from "hardhat";
-import { StakedCelo__factory } from "../typechain-types/factories/StakedCelo__factory";
-import { StakedCelo } from "../typechain-types/StakedCelo";
 import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
 import { randomAddress, randomSigner } from "./utils";
 import { parseUnits } from "ethers/lib/utils";
+import { Contract, ContractFactory } from "ethers";
+
+interface ProxyDeployTestData {
+  contractName: string;
+  fixtureName: string;
+  initializeArgs: any[];
+  ownerIsDeployer?: boolean;
+}
+
+const tests: ProxyDeployTestData[] = [
+  {
+    contractName: "StakedCelo",
+    fixtureName: "TestStakedCelo",
```

Comment:
Can these be done using the real deployments rather than test fixtures? Then probably wouldn't need a `fixtureName` in these parameters, just deploy `core` in a `beforeEach`.

Reply:
I used core and since it means that owner will be multisig I also added proposal handling